### DISCUSSION TITLE
Mask PowerShell statement

### DISCUSSION THREAD

**Discussion by VoidMonk** [community/community#25859] [Actions]
<p>Hi, loving GitHub Actions!</p>
<p>Quick question: a step in my workflow job has a PowerShell statement that reads a long value from the deployment payload, and concatenates it to make a variable value. In the logs, the <code>payload.long_value</code> is printed, but I’d like to mask it.</p>
<pre><code>shell: powershell
run: |
    $something_long = "something.." + "${{ github.event.deployment.payload.long_value }}"
</code></pre>
<p>I can’t put it in an environment variable because the value is too long. Is there any way to mask a statement or payload property in PowerShell code?</p>

**Comment by VoidMonk** [comment]
<p>I tried using the <a href="https://help.github.com/en/articles/development-tools-for-github-actions#stop-and-start-log-commands-stop-commands" rel="nofollow noopener">Stop and start log commands</a> in our PowerShell inline-code as below:</p>
<pre><code>shell: powershell
run: |
echo ::stop-commands::pause-logging
$something_long = "something.." + "${{ github.event.deployment.payload.long_value }}"
echo ::pause-logging::
</code></pre>
<p>But it still shows that statement with <code>payload.long_value</code> in the logs.</p>
<p>If there any way to hide/mask it in the logs?</p>

**Comment by VoidMonk** [comment]
<p>This is crucial for us, in terms of data security, to consider Actions for production CI/CD. Please advise <a class="mention" href="/ethomson">@ethomson</a> <a class="mention" href="/chrispat">@chrispat</a> <a class="mention" href="/mscoutermarsh">@mscoutermarsh</a> </p>

**Comment by chrispat** [comment, marked as answer]
<p>If the value is too long for an environment variable we are not going to be able to mask it.  My suggestion would be to read the value from the payload file on disk rather than using the context in this case.  You can find the path to the  payload file in the GITHUB_EVENT_PATH environment variable <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#default-environment-variables</a>.</p>

**Comment by VoidMonk** [comment]
<p>Thank you! (Btw, variable name is GITHUB_EVENT_PATH)</p>
